### ISSUE TITLE
Compiling on ARM64 Manjaro Linux fails

### ISSUE THREAD

**Issue by CanadianPinguin** [rethinkdb/rethinkdb#6999]
Trying to build from source on Manjaro(5.10.76-2-MANJARO-ARM) fails at v8 :
```
[430/457] BUILD v8_3.30.33.16-patched
.././src/base/platform/time.h:337:14: note: because ‘v8::base::TimeTicks’ has user-provided ‘v8::base::TimeTicks& v8::base::TimeTicks::operator=(v8::base::TimeTicks)’
  337 |   TimeTicks& operator=(const TimeTicks other) {
      |              ^~~~~~~~
In file included from ../src/mksnapshot.cc:15:
.././src/assembler.h: In static member function ‘static void v8::internal::ExternalReference::set_redirector(v8::internal::Isolate*, void* (*)(void*, v8::internal::ExternalReference::Type))’:
.././src/assembler.h:984:9: warning: cast between incompatible function types from ‘void* (*)(void*, v8::internal::ExternalReference::Type)’ to ‘void* (*)()’ [-Wcast-function-type]
  984 |         reinterpret_cast<ExternalReferenceRedirectorPointer*>(redirector));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.././src/assembler.h: In static member function ‘static void* v8::internal::ExternalReference::Redirect(v8::internal::Isolate*, v8::internal::Address, v8::internal::ExternalReference::Type)’:
.././src/assembler.h:997:9: warning: cast between incompatible function types from ‘void* (*)()’ to ‘void* (*)(void*, v8::internal::ExternalReference::Type)’ [-Wcast-function-type]
  997 |         reinterpret_cast<ExternalReferenceRedirector*>(
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  998 |             isolate->external_reference_redirector());
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /usr/bin/c++ -pthread -ldl -o /home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/mksnapshot -Wl,--start-group /home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/obj.target/mksnapshot/src/mksnapshot.o /home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/obj.target/tools/gyp/libv8_base.a /home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/obj.target/tools/gyp/libv8_nosnapshot.a /home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/obj.target/tools/gyp/libv8_libplatform.a /home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/obj.target/third_party/icu/libicui18n.a /home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/obj.target/third_party/icu/libicuuc.a /home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/obj.target/tools/gyp/libv8_libbase.a /home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/obj.target/third_party/icu/libicudata.a -Wl,--end-group -lrt
  LD_LIBRARY_PATH=/home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/lib.host:/home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/lib.target:$LD_LIBRARY_PATH; export LD_LIBRARY_PATH; cd ../tools/gyp; mkdir -p /home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/obj.target/v8_snapshot/geni; "/home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/mksnapshot" --log-snapshot-positions --logfile "/home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/obj.target/v8_snapshot/geni/snapshot.log" --random-seed 314159265 "/home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/obj.target/v8_snapshot/geni/snapshot.cc"
make[3]: *** [tools/gyp/v8_snapshot.target.arm64.release.mk:13: /home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out/arm64.release/obj.target/v8_snapshot/geni/snapshot.cc] Segmentation fault (core dumped)
make[3]: Leaving directory '/home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build/out'
make[2]: *** [Makefile:286: arm64.release] Error 2
make[2]: Leaving directory '/home/user/rethinkdb-2.4.1/build/external/v8_3.30.33.16-patched/build'

Full error log: ./build/external/v8_3.30.33.16-patched_install.log
make[1]: *** [mk/support/build.mk:149: build/external/v8_3.30.33.16-patched/lib/libv8.a] Error 1
make: *** [Makefile:53: make] Error 2

```

**Comment by srh** [comment]
This might be fixed in v2.4.x by replacing v8 with QuickJS.  I have not tested QuickJS on ARM yet.

**Comment by srh** [comment]
v2.4.x is building on ARM64.  So this has probably been fixed on Manjaro with the QuickJS port.  Tested on Ubuntu Focal and Debian Bullseye.  If it's failing for you on another build step, please let us know.

Closing.
